CAP/gap/PrintingFunctions.gi

```diff
@@ -43,7 +44,13 @@ InstallGlobalFunction( InfoStringOfInstalledOperationsOfCategory,
         od;
     fi;
     
-    list_of_non_algorithmic_mathematical_properties := Difference( list_of_mathematical_properties, list_of_algorithmic_properties );
+    list_of_non_algorithmic_mathematical_properties_implied_by_algorithmic_properties :=
+      Difference( Concatenation( List( list_of_algorithmic_properties, p -> ListImpliedFilters( ValueGlobal( p ) ) ) ),
+              list_of_potential_algorithmic_properties );
```

Comment:
Ah, now I finally understand the logic.

1. As it is, this list should be called `list_of_potential_non_algorithmic_mathematical_properties_implied_by_algorithmic_properties`.
2. Removing `list_of_potential_algorithmic_properties` here does not make a difference for the result of `list_of_non_algorithmic_mathematical_properties`, right? So we can simplify this and avoid the call to `Difference`.

I also suggest to include the filter itself in the output of `ListImpliedFilters`. This makes more sense mathematically (the filter certainly implies itself) and allows to simplify the code here and at other places.

Reply:
> 1. As it is, this list should be called `list_of_potential_non_algorithmic_mathematical_properties_implied_by_algorithmic_properties`.

No, this is the list of the non-algorithmic properties *implied* by the algorithmic ones, so they are not "potential" properties, but actual properties.

> 2. Removing `list_of_potential_algorithmic_properties` here does not make a difference for the result of `list_of_non_algorithmic_mathematical_properties`, right? So we can simplify this and avoid the call to `Difference`.

No, it does make a difference: A \ ( B \ C) is not the same as A \ B.

> I also suggest to include the filter itself in the output of `ListImpliedFilters`. This makes more sense mathematically (the filter certainly implies itself) and allows to simplify the code here and at other places.

Yes, good idea, but in this particular case it wouldn't make a difference. I can still do this independent of this commit.

Reply:
Ad 1.: Ah, correct, sorry.

Ad 2.: It does make a difference for `list_of_non_algorithmic_mathematical_properties_implied_by_algorithmic_properties` but it does not make a difference for `list_of_non_algorithmic_mathematical_properties` if I am not mistaken.

Ad `ListImpliedFilters`.: It would make a difference: if we remove the `Difference` in my point 2. and modify `ListImpliedFilters`, then the Concatenation in `list_of_potential_algorithmic_properties := ...` could be dropped and thus the code would be much simpler.

Reply:
>>> Removing `list_of_potential_algorithmic_properties` here does not make a difference for the result of `list_of_non_algorithmic_mathematical_properties`, right? So we can simplify this and avoid the call to Difference.

>> No, it does make a difference: A \ ( B \ C ) is not the same as A \ B.

> Ad 2.: It does make a difference for `list_of_non_algorithmic_mathematical_properties_implied_by_algorithmic_properties` but it does not make a difference for `list_of_non_algorithmic_mathematical_properties` if I am not mistaken.

Let be more specific:

Currently:

* `list_of_non_algorithmic_mathematical_properties_implied_by_algorithmic_properties` = IP \ PAP
* `list_of_non_algorithmic_mathematical_properties` = MP \ ( AP u ( IP \ PAP ) )

If I understand your suggestion correctly it is:

* `list_of_non_algorithmic_mathematical_properties_implied_by_algorithmic_properties` = IP
* `list_of_non_algorithmic_mathematical_properties` = MP \ ( AP u IP )

But this is not the same because now `list_of_non_algorithmic_mathematical_properties` also includes the intersection of MP and PAP, which it did not include before.

Reply:
> Ad `ListImpliedFilters`.: It would make a difference: if we remove the `Difference` in my point 2. and modify `ListImpliedFilters`, then the Concatenation in `list_of_potential_algorithmic_properties := ...` could be dropped and thus the code would be much simpler.

I don't understand this point, probably you are arguing based on your suggestion. Let us clarify point 2. first.

Reply:
A good example to test your suggestion is `examples/TerminalCategoryWithSingleObject.g`.

Reply:
I pushed what we discussed yesterday.